### PR TITLE
Handling errors in Remix

### PR DESCRIPTION
**Pull request by saliougaye**

### CODE REVIEW

File: app/lib/matador/navigations.tsx [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+  },
+  {
+    label: "Clients",
+    href: "./clients",
+    color: "green",
+    icon: <BsArrowsAngleContract size={20} />,
+  },
+];
+
+export default Navigations;
```
Comment (nullndr) [inline app/lib/matador/navigations.tsx:27]: I thought about the navigation, and I think this should be placed in `app/lib/matador/helpers/ui-helpers.ts`

File: app/routes/matador.tsx [hunk 2]
```diff
@@ -1,18 +1,12 @@
...
 import { useEffect, useState } from "react";
-import { BiInfoCircle } from "react-icons/bi";
-import { BsArrowsAngleContract } from "react-icons/bs";
-import { DiRedis } from "react-icons/di";
-import type { NavbarItem } from "~/lib/matador/components";
-import { NavBar } from "~/lib/matador/components/navbar";
+import { NavBar, ErrorFallback } from "~/lib/matador/components/index";
 import { themeKeyLocalStorage } from "~/lib/matador/constants";
 import { getMatadorVersion } from "~/lib/matador/index.server";
+import Navigations from "~/lib/matador/navigations";
```
Comment (nullndr) [inline app/routes/matador.tsx:9]: remember to refactor this like above

File: app/routes/matador/$queueName/index.tsx [hunk 3]
```diff
@@ -1,14 +1,16 @@
...
 import React, { useState } from "react";
 import { JobsTable } from "~/lib/matador/components/jobs-table";
 import { StatCard } from "~/lib/matador/components/stat-card";
 import type { BullJob } from "~/lib/matador/index.server";
 import { getQueueJobs, getQueues } from "~/lib/matador/index.server";
-import type { JobStatus} from "~/lib/matador/types/JobStatus";
+import type { JobStatus } from "~/lib/matador/types/JobStatus";
 import { JobStatuses } from "~/lib/matador/types/JobStatus";
 
+import { ErrorFallback } from "~/lib/matador/components/error";
```
Comment (nullndr) [inline app/routes/matador/$queueName/index.tsx:12]: Is there a blank line?

File: app/routes/matador/$queueName/repeat.$jobId.tsx [hunk 4]
```diff
@@ -1,16 +1,16 @@
...
 import type { BullJob, Job } from "~/lib/matador/index.server";
-import {
-  getRepeatableQueueJobs
-} from "~/lib/matador/index.server";
-import type { JobStatus} from "~/lib/matador/types/JobStatus";
+import { getRepeatableQueueJobs } from "~/lib/matador/index.server";
+import type { JobStatus } from "~/lib/matador/types/JobStatus";
 import { JobStatuses } from "~/lib/matador/types/JobStatus";
 
+import { ErrorFallback } from "~/lib/matador/components/error";
```
Comment (nullndr) [inline app/routes/matador/$queueName/repeat.$jobId.tsx:12]: Is there a blank line?

File: app/routes/matador/index.tsx [hunk 5]
```diff
@@ -1,5 +1,7 @@
 import { Grid, Text, Title } from "@mantine/core";
+import { useCatch } from "@remix-run/react";
 
+import { ErrorFallback } from "~/lib/matador/components/error";
 export default function Index() {
```
Comment (nullndr) [inline app/routes/matador/index.tsx:5]:
```suggestion
import { useCatch } from "@remix-run/react";
import { ErrorFallback } from "~/lib/matador/components/error";

export default function Index() {
```

File: app/routes/matador/queues.tsx [hunk 6]
```diff
@@ -1,17 +1,19 @@
...
+import { NavLink, useCatch, useLoaderData } from "@remix-run/react";
 import { Link } from "~/lib/matador/components/ui/Link";
 import type { RedisInfo } from "~/lib/matador/index.server";
 import { getQueues, getRedisInfo } from "~/lib/matador/index.server";
 import { StatCard } from "../../lib/matador/components/stat-card";
 
 type LoaderData = { queues: string[]; serverInfo: RedisInfo };
+import { ErrorFallback } from "~/lib/matador/components/error";
 
 export const loader: LoaderFunction = async ({
```
Comment (nullndr) [inline app/routes/matador/queues.tsx:18]:
```suggestion
import { StatCard } from "../../lib/matador/components/stat-card";
import { ErrorFallback } from "~/lib/matador/components/error";

type LoaderData = { queues: string[]; serverInfo: RedisInfo };

export const loader: LoaderFunction = async ({
```

File: app/routes/matador/redis.tsx [hunk 7]
```diff
@@ -1,11 +1,13 @@
...
-import { useLoaderData } from "@remix-run/react";
+import { useCatch, useLoaderData } from "@remix-run/react";
 import React from "react";
 import { StatCard } from "~/lib/matador/components/stat-card";
 import type { RedisInfo } from "~/lib/matador/index.server";
 import { getRedisInfo } from "~/lib/matador/index.server";
 
+import { ErrorFallback } from "~/lib/matador/components/error";
+
 type LoaderData = RedisInfo;
```
Comment (nullndr) [inline app/routes/matador/redis.tsx:11]:
```suggestion
import { getRedisInfo } from "~/lib/matador/index.server";
import { ErrorFallback } from "~/lib/matador/components/error";

type LoaderData = RedisInfo;
```

**Review by nullndr — CHANGES_REQUESTED** [review]
🛑

**Comment by nullndr** [comment]
If you use VSCode please install the Prettier plugin to auto format the document on save

File: app/lib/matador/navigations.tsx (hunk 1 above)
Comment (saliougaye) [reply app/lib/matador/navigations.tsx]: Fixed with f089a705ecee1a85e45b3512238c5af13eef525a

File: app/routes/matador/$queueName/index.tsx (hunk 3 above)
Comment (saliougaye) [reply app/routes/matador/$queueName/index.tsx]: Fixed with f089a705ecee1a85e45b3512238c5af13eef525a

File: app/routes/matador/$queueName/repeat.$jobId.tsx (hunk 4 above)
Comment (saliougaye) [reply app/routes/matador/$queueName/repeat.$jobId.tsx]: Fixed with f089a705ecee1a85e45b3512238c5af13eef525a

File: app/routes/matador/index.tsx (hunk 5 above)
Comment (saliougaye) [reply app/routes/matador/index.tsx]: Fixed with f089a705ecee1a85e45b3512238c5af13eef525a

File: app/routes/matador/queues.tsx (hunk 6 above)
Comment (saliougaye) [reply app/routes/matador/queues.tsx]: Fixed with f089a705ecee1a85e45b3512238c5af13eef525a

File: app/routes/matador/redis.tsx (hunk 7 above)
Comment (saliougaye) [reply app/routes/matador/redis.tsx]: Fixed with f089a705ecee1a85e45b3512238c5af13eef525a

**Review by nullndr — APPROVED** [review]
🚀 🚀 🚀

File: app/routes/matador.tsx (hunk 2 above)
Comment (saliougaye) [reply app/routes/matador.tsx]: Fixed with f089a705ecee1a85e45b3512238c5af13eef525a

**Pull request merged**